scripts/lib/email-mirror-server

```diff
@@ -0,0 +1,182 @@
+        from_email=settings.NOREPLY_EMAIL_ADDRESS,
+        to=[settings.ZULIP_ADMINISTRATOR],
+    )
+    mail.attach(None, msg, "message/rfc822")
+    try:
+        mail.send()
+    except smtplib.SMTPResponseException as e:
+        logger.exception(
+            "Error sending bounce email to %s with error code %s: %s",
+            mail.to,
+            e.smtp_code,
+            e.smtp_error,
+            stack_info=True,
+        )
+    except smtplib.SMTPException as e:
+        logger.exception("Error sending bounce email to %s: %s", mail.to, str(e), stack_info=True)
+
+
+class MessageHandler(BaseMessageHandler):
+    def __init__(self) -> None:
+        super().__init__(email.message.Message)
+
+    async def handle_RCPT(
+        self, server: SMTP, session: Session, envelope: Envelope, address: str, rcpt_options: Any
+    ) -> str:
+        # Rewrite all postmaster email addresses to just "postmaster"
+        if address.lower() == "postmaster":
+            envelope.rcpt_tos.append("postmaster")
+            return "250 Continue"
+        try:
+            if get_email_gateway_message_string_from_address(address).lower() == "postmaster":
+                envelope.rcpt_tos.append("postmaster")
+                return "250 Continue"
+        except ZulipEmailForwardError:
+            pass
+
+        try:
+            await sync_to_async(validate_to_address)(address)
+            if not is_missed_message_address(address):
+                # Missed message addresses are one-time use, so we don't need
+                # to worry about emails to them resulting in message spam.
+                recipient_realm = sync_to_async(decode_stream_email_address)(address)[0].realm
+                rate_limit_mirror_by_realm(recipient_realm)
+        except RateLimited:
+            logger.warning(
+                "Rejecting a MAIL FROM: %s to realm: %s - rate limited.",
+                envelope.mail_from,
+                recipient_realm.name,
+            )
+            return "550 5.1.1 Rate-limited due to too many emails on this realm."
+
+        except ZulipEmailForwardError as e:
+            return f"550 5.1.1 Bad destination mailbox address: {e}"
+
+        envelope.rcpt_tos.append(address)
+        return "250 Continue"
+
+    async def handle_DATA(self, server: SMTP, session: Session, envelope: Envelope) -> str:
+        message = self.prepare_message(session, envelope)
+
+        msg_base64 = base64.b64encode(bytes(message))
+        for address in envelope.rcpt_tos:
+            if address == "postmaster":
+                send_to_postmaster(message)
+            else:
+                await sync_to_async(queue_json_publish)(
+                    "email_mirror",
+                    {
+                        "rcpt_to": address,
+                        "msg_base64": msg_base64.decode(),
+                    },
+                )
+
+        return "250 OK"
+
+
+class Controller(BaseController):
```

Comment:
It seems silly to subclass this just to call the constructor with some options, when we could call the constructor with some options anyway? Especially since the renamed import of `aiosmtpd.controller.Controller` as `BaseController` is confusable with `aiosmtpd.controller.BaseController`.